src/cargo/util/interning.rs

```diff
@@ -90,6 +90,12 @@ impl InternedString {
     pub fn as_str(&self) -> &'static str {
         self.inner
     }
+
+    /// A faster implementation of hash that is completely compatible with HashMap,
+    /// but does not have a stable value between runs of the program.
+    pub fn fast_hash<H: std::hash::Hasher>(&self, state: &mut H) {
```

Comment:
Not quite compatible with `HashMap` because a lookup with a `""` will always fail because this diverges in the hash from `&str`

Reply:
I will try and update comment. It is not safe to use this hash for intern string in general while it has the deref to `str`. But, I think it's safe in this case. I need to articulate when it is safe to use it.